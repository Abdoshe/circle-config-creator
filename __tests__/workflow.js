/* @flow */

import Workflow from '../src/workflow';
import Job from '../src/job';
import Branches from '../src/branches';

function p(obj) {
  // eslint-disable-next-line no-console
  console.log(JSON.stringify(obj, null, 2));
}

describe('Workflow', () => {
  it('can create a simple workflow', () => {
    const w = new Workflow('test-workflow');

    const job = new Job('test-job');

    expect(w.job(job).compose()).toMatchSnapshot();
  });

  it('can require another job to run', () => {
    const w = new Workflow('test-workflow');

    const job = new Job('test-job');
    const requires = new Job('required-by-test-job');

    expect(w.job(job, [requires]).compose()).toMatchSnapshot();
  });

  it('can take triggers', () => {
    const branches = new Branches().only('test-branch');

    const w = new Workflow('test-workflow');
    w.schedule('* * * * *', branches);

    expect(w.compose()).toMatchSnapshot();
  });
});

