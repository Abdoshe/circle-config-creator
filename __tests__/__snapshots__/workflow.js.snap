// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workflow can create a simple workflow 1`] = `
Object {
  "test-workflow": Object {
    "jobs": Array [
      "test-job",
    ],
  },
}
`;

exports[`Workflow can require another job to run 1`] = `
Object {
  "test-workflow": Object {
    "jobs": Array [
      Object {
        "test-job": Object {
          "requires": Array [
            "required-by-test-job",
          ],
        },
      },
    ],
  },
}
`;

exports[`Workflow can take triggers 1`] = `
Object {
  "test-workflow": Object {
    "jobs": Array [],
    "triggers": Array [
      Object {
        "schedule": Object {
          "cron": "* * * * *",
          "filters": Object {
            "branches": Object {
              "only": Array [
                "test-branch",
              ],
            },
          },
        },
      },
    ],
  },
}
`;
